internal/handlers/pg/pgdb/query.go

```diff
@@ -251,12 +251,13 @@ func prepareWhereClause(sqlFilters *types.Document) (string, []any, error) {
 			return "", nil, lazyerrors.Error(err)
 		}
 
-		if k != "" {
+		switch {
```

Comment:
I like this switch-case approach, it looks more readable than a chain of `if`s!

Reply:
It is, but do we really need that chain? The first condition looks sus, and if it's not actually needed, that's not a chain anymore

Reply:
`NewPathFromString`  won't panic, it returns `ErrDocumentPathEmptyKey` error. So that check now can be removed :hugs: 

`k[0] == '$'` can be replaced with `strings.HasPrefix(k,'$')` to avoid panic too.

Reply:
It looks fixed to me!